typegate/src/runtimes/random.ts

```diff
@@ -60,33 +71,45 @@ export class RandomRuntime extends Runtime {
 
   execute(typ: TypeNode): Resolver {
     return () => {
-      const config = typ.config ?? {};
-      if (Object.prototype.hasOwnProperty.call(config, "gen")) {
-        const { gen, ...arg } = config;
-        return this.chance[gen as string](arg);
-      }
-
-      switch (typ.type) {
-        case "object":
-          return {};
-        case "array":
-          // TODO
-          return [];
-        case "integer":
-          return this.chance.integer();
-        case "string":
-          if (typ.format === "uuid") {
-            return this.chance.guid();
-          }
-          if (typ.format === "email") {
-            return this.chance.email();
+      const randomizeRecursively = (typ: TypeNode): any => {
+        const config = typ.config ?? {};
+        if (Object.prototype.hasOwnProperty.call(config, "gen")) {
+          const { gen, ...arg } = config;
+          return this.chance[gen as string](arg);
+        }
+        switch (typ.type) {
+          case "object":
+            return {};
+          case "integer":
+            return this.chance.integer();
+          case "string":
+            if (typ.format === "name") {
+              return this.chance.name();
+            }
+            if (typ.format === "uuid") {
+              return this.chance.guid();
+            }
+            if (typ.format === "email") {
+              return this.chance.email();
+            }
+            return this.chance.string();
+          case "boolean":
+            return this.chance.bool();
+          case "array": {
+            const res = [];
+            let size = this.chance.integer({ min: 1, max: 10 });
+            while (size--) {
+              const childNodeName = this.getTgTypeNameByIndex(typ.items);
```

Comment:
Calls `this.getTgTypeNameByIndex` on each iteration.

Reply:
You are right, I will refactor fact.